packages/powersync/lib/src/connector.dart

```diff
@@ -113,7 +113,12 @@ class PowerSyncCredentials {
     try {
       List<String> parts = token.split('.');
       if (parts.length == 3) {
-        final rawData = base64Decode(parts[1]);
+        String part = parts[1];
+        // dart:convert doesn't like missing padding
+        if (part.length % 4 > 0) {
+          part += '=' * (4 - part.length % 4);
+        }
+        final rawData = base64Decode(part);
```

Comment:
The issue is that JWTs use base64-url encoding instead of standard base64. Apart from the padding, there are also other minor encoding differences. Could you check if this works?

```suggestion
        final rawData = base64Url.decode(part);
```

Reply:
I'll test, but their docs suggest otherwise: https://api.flutter.dev/flutter/dart-convert/base64Url-constant.html

Reply:
I tested with your suggested change and still get the same FormatException about invalid length.

https://github.com/dart-lang/sdk/issues/39510 seems to suggest this is deliberate.

Reply:
However, there does seem to be a better way to do this, brb

Reply:
using https://api.flutter.dev/flutter/dart-convert/Base64Codec/normalize.html seems like a better approach

Reply:
Thanks! Turns out my initial suggested change doesn't even make a difference - `base64url.decode` and `base64.decode` does the same thing.